code/ui/manager/src/components/sidebar/Refs.tsx

```diff
@@ -131,6 +131,12 @@ export const Ref: FC<RefType & RefProps & { status?: State['status'] }> = React.
   const state = getStateType(isLoading, isAuthRequired, isError, isEmpty);
   const [isExpanded, setExpanded] = useState<boolean>(expanded);
 
+  useEffect(() => {
+    if (index && selectedStoryId && index[selectedStoryId]) {
+      setExpanded(true);
+    }
+  }, [setExpanded, index, selectedStoryId]);
```

Comment:
I'm not sure we really need the useEffect?

Would this work instead:
```
const [isExpanded, setExpanded] = useState<boolean>(expanded || index && (selectedStoryId && index[selectedStoryId]));
```

Reply:
`useEffect` is needed to expand the Ref when storybook is starting up, since `index` is undefined on the component's first render.

I also used `useEffect` to update `isExpanded` when the user selects a new story using a keyboard shortcut or the shortcuts menu, so that it acts like the `Tree` component used in the sidebar. The `useEffect` also makes it so that `index[selectedStoryId]` is only checked when either `selectedStoryId` or `index` is updated.

p.s. I added some gifs to the PR description to make clearer how the Ref acts now.